spark/dl/src/main/scala/com/intel/analytics/bigdl/models/resnet/TrainImageNet.scala

```diff
@@ -68,31 +68,25 @@ object TrainImageNet {
       val model = if (param.modelSnapshot.isDefined) {
         Module.load[Float](param.modelSnapshot.get)
       } else {
-        Engine.getEngineType() match {
-          case MklBlas =>
-            val curModel =
-              ResNet(classNum = param.classes, T("shortcutType" -> shortcut, "depth" -> param.depth,
-                "optnet" -> param.optnet, "dataSet" -> dataSetType))
-            if (param.optnet) {
-              ResNet.shareGradInput(curModel)
-            }
-            ResNet.modelInit(curModel)
-
-            /* Here we set parallism specificall for BatchNormalization and its Sub Layers, this is
-            very useful especially when you want to leverage more computing resources like you want
-            to use as many cores as possible but you cannot set batch size too big for each core due
-            to the memory limitation, so you can set batch size per core smaller, but the smaller
-            batch size will increase the instability of convergence, the synchronization among BN
-            layers basically do the parameters synchronization among cores and thus will avoid the
-            instability while improves the performance a lot. */
-            val parallisim = Engine.coreNumber
-            setParallism(curModel, parallisim)
-
-            curModel
-          case MklDnn =>
-            nn.mkldnn.ResNet.graph(param.batchSize / Engine.nodeNumber(), param.classes,
-              T("depth" -> 50, "dataSet" -> ImageNet))
+        val curModel =
+          ResNet(classNum = param.classes, T("shortcutType" -> shortcut, "depth" -> param.depth,
+            "optnet" -> param.optnet, "dataSet" -> dataSetType))
+        if (param.optnet) {
+          ResNet.shareGradInput(curModel)
         }
+        ResNet.modelInit(curModel)
+
+        /* Here we set parallism specificall for BatchNormalization and its Sub Layers, this is
+        very useful especially when you want to leverage more computing resources like you want
+        to use as many cores as possible but you cannot set batch size too big for each core due
+        to the memory limitation, so you can set batch size per core smaller, but the smaller
+        batch size will increase the instability of convergence, the synchronization among BN
+        layers basically do the parameters synchronization among cores and thus will avoid the
+        instability while improves the performance a lot. */
+        val parallisim = Engine.coreNumber
+        setParallism(curModel, parallisim)
```

Comment:
I don't think we need this for MKL-DNN as we use all assigned cores

Reply:
Yeah, when training with MKLDNN, `setParallism` is useless even through it is set, but for MKL-BLAS, it can speed up. So I just keep this line .

Reply:
it's better to have EngineType checking so that only MKL-BLAS mode will be using it

Reply:
ok, changed